kernels/portable/cpu/util/broadcast_util.cpp

```diff
@@ -213,10 +213,20 @@ ET_NODISCARD Error get_broadcast_target_size(
     Tensor::SizesType* out_sizes,
     const size_t out_sizes_len,
     size_t* out_dim) {
-  ET_CHECK_OR_RETURN_ERROR(
-      tensors_are_broadcastable_between(a_size, b_size),
-      InvalidArgument,
-      "Two input tensors should be broadcastable.\n");
+  if (!tensors_are_broadcastable_between(a_size, b_size)) {
+    const auto a_shape_str = tensor_shape_to_c_string(
```

Comment:
This new API is great, keeping the caller totally ignorant of how big a buffer to allocate.